src/main/java/com/example/CleanFolders.java

```diff
@@ -0,0 +1,127 @@
+package com.example;
+
+import java.io.*;
+import java.util.*;
+
+/**
+ * Egy könyvtárstruktúrában dokumentumokat szerkesztenek.
+ *
+ * Minden dokumentumszerkesztéskor .bak kiterjesztésű fájlok biztonsági mentés jönnek létre
+ * (doc1.doc-hoz doc1.bak), azonban gyakori, hogy a szerkesztő bezárása után ottmaradnak.
+ *
+ * Sok esetben a dokumentumokat már törlik is, és üres könyvtárra egyébként nem lenne szükség,
+ * de a .bak fájlok miatt a könyvtárak megmaradnak.
+ *
+ * A feladat egy olyan program készítése, ami a paraméterként kapott könyvtárból kiindulva
+ * kitisztítja a dokumentum nélkül ottmaradt .bak fájlokat, és az üres könyvtárakat is törli.
+ *
+ * A struktúra egy nagyon nagy (több milliós) virtuális fájlrendszer, és egy törlés viszonylag lassú művelet,
+ * ezért fontos több szálon végezni a törléseket - a fájlrendszer elosztott,
+ * ezért jól fogja bírni a párhuzamos terhelést.
+ *
+ * Megoldásként Java nyelvű forráskódot várunk (tetszőleges Java verzió használható).
+ */
+public class CleanFolders {
+	private static final List<File> deletableBaks = new ArrayList<>();
+	private static final List<File> folders = new ArrayList<>();
+
+	/**
+	 * Detects directories under the given directory and collects all .bak files without corresponding .doc files next to them.
+	 * The method uses checkFolder() for detection and loneBakDelete() for deletion.
+	 * @param startFolder	File of the starting directory.
+	 */
+	public static void bakDeletion(File startFolder) {
+		checkFolder(startFolder);
+		loneBakDelete();
+	}
+
+	/**
+	 * Returns the collected, deletable .bak files.
+	 * Created for opening a safe backdoor for testing purposes.
+	 * @return	List<File> of .bak files.
+	 */
+	public static List<File> getDeletableBaks() {
+		return deletableBaks;
+	}
+
+	/**
+	 * Returns the collected directories.
```

Comment:
I think this is a great example of a not useful comment. What are the collected directories here?

Reply:
Hmm. The "Returns all folders." would not give better satisfaction, I guess. The collector method literally collects all folders under the given root. I modified Javadocs for the collector and the 2 getter methods with new sentences and referencing to each other for better understanding, please check when pushed.